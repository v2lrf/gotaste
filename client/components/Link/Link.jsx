import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

function Link({ href, children, className, onClick }) {
  const linkClasses = classnames('text-red-dark no-underline hover:underline', {
    [className]: className
  })
  return (
    <a href={href} className={linkClasses} onClick={onClick}>
      {children}
    </a>
  )
}

Link.propTypes = {
  href: PropTypes.string,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func
}

Link.defaultProps = {
  href: '#',
  className: null,
  onClick: null
}

export default Link
