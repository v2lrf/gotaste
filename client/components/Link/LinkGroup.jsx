import React from 'react'
import PropTypes from 'prop-types'

function LinkGroup({ label, children }) {
  return (
    <div className="text-right mb-4">
      {label && (
        <span className="mr-2 text-grey-dark text-sm font-bold uppercase">
          {label}
        </span>
      )}
      {children}
    </div>
  )
}

LinkGroup.propTypes = {
  label: PropTypes.string,
  children: PropTypes.node.isRequired
}

LinkGroup.defaultProps = {
  label: null
}

export default LinkGroup
