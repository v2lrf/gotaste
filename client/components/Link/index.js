export { default as LinkGroup } from './LinkGroup'
export { default } from './Link'
