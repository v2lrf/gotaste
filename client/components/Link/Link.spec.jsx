import React from 'react'
import Link from './Link'

describe('Link', () => {
  let component
  beforeEach(() => {
    component = shallow(<Link>Some content!</Link>)
  })

  describe('Snapshots', () => {
    it('should match the default snapshot', () => {
      expect(component).toMatchSnapshot()
    })

    it('should match the snapshot with custom href', () => {
      component.setProps({ href: '/some-path' })
      expect(component).toMatchSnapshot()
    })

    it('should match the snapshot with custom className', () => {
      component.setProps({ className: 'foo bar' })
      expect(component).toMatchSnapshot()
    })
  })

  describe('Actions', () => {
    it('should call the onClick handler when clicked', () => {
      const onClickHandler = jest.fn()
      component.setProps({ onClick: onClickHandler })
      component.simulate('click')
      expect(onClickHandler).toHaveBeenCalled()
    })
  })
})
