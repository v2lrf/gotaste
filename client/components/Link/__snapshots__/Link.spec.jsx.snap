// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Snapshots should match the default snapshot 1`] = `
<a
  className="text-red-dark no-underline hover:underline"
  href="#"
  onClick={null}
>
  Some content!
</a>
`;

exports[`Link Snapshots should match the snapshot with custom className 1`] = `
<a
  className="text-red-dark no-underline hover:underline foo bar"
  href="#"
  onClick={null}
>
  Some content!
</a>
`;

exports[`Link Snapshots should match the snapshot with custom href 1`] = `
<a
  className="text-red-dark no-underline hover:underline"
  href="/some-path"
  onClick={null}
>
  Some content!
</a>
`;
